google/cloud/sql/connector/utils.py

```diff
@@ -14,13 +14,25 @@
 limitations under the License.
 """
 
+import asyncio
+from functools import partial, wraps
 
 import pymysql.cursors
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
 from cryptography.hazmat.primitives.asymmetric import rsa
 
 
+def run_function_as_async(func):
+    @wraps(func)
+    async def wrapped_sync_function(*args, **kwargs):
+        partial_func = partial(func, *args, **kwargs)
+        loop = asyncio.get_event_loop()
+        return await loop.run_in_executor(None, partial_func)
```

Comment:
are we using the default executor for the rest of the threads? (if not, maybe we should?)

Reply:
I think this is the only `run_in_executor` call